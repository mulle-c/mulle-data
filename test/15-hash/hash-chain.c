#include <mulle-data/mulle-data.h>
#include <stdio.h>


int   main( void)
{
   uintptr_t  hash;
   void       *state;

   // leftovers
   hash = _mulle_hash_32( "VfL Bochum 1848", 15);
   printf( "32: %#016tx\n", hash);

   hash = _mulle_hash_64( "VfL Bochum 1848", 15);
   printf( "64: %#016tx\n", hash);

   // these hash values must be identical to above, otherwise chaining
   // is broken
   state = NULL;
          mulle_hash_chained_32( "VfL Bochum", 10, &state);
          mulle_hash_chained_32( " 1848", 5, &state);
   hash = mulle_hash_chained_32( NULL, 0, &state);
   printf( "chain32: %#016tx\n", hash);

   state = NULL;
          mulle_hash_chained_64( "VfL Bochum", 10, &state);
          mulle_hash_chained_64( " 1848", 5, &state);
   hash = mulle_hash_chained_64( NULL, 0, &state);
   printf( "chain64: %#016tx\n", hash);

   return( 0);
}
