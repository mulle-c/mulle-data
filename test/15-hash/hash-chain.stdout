32: 0x000000fc64daae
64: 0x1d20234e9e711e26
chain32: 0x000000fc64daae
chain64: 0x1d20234e9e711e26
